docs/source/extending/PluginFramework.rst

```diff
@@ -0,0 +1,138 @@
+MSTICPy Plugin Framework
+========================
+
+MSTICPy has several extensibility points, where you
+can create your own modules to support data and
+enrichment sources.
+
+These currently include:
+
+- Data Providers
+- Threat Intelligence Providers
+- Context Providers
+
+If you create one of these and think that it would be useful
+to other users, please consider contributing it to MSTICPy.
```

Comment:
Maybe we could link to contribution guidance here or even make the language a bit stronger such as "If you are creating an extension that others can use consider contributing it to MSTICPy, however if you are creating something specific to you or your organization then you can create local provider modules.